### PR TITLE
Add auth errors

### PR DESCRIPTION
**Pull request by sisuresh**

### CODE REVIEW

File: Stellar-contract.x [hunk 1]
```diff
@@ -128,6 +129,16 @@ enum SCHostStorageErrorCode
     HOST_STORAGE_GET_ON_DELETED_KEY = 5
 };
 
+enum SCHostAuthErrorCode
+{
+    HOST_AUTH_UNKNOWN_ERROR = 0,
+    HOST_AUTH_WRONG_NONCE = 1,
```
Comment (dmkozh) [inline Stellar-contract.x]: Can we rename this to just `NONCE_ERROR` in order to not touch it anymore when we change the nonce approach?

File: Stellar-contract.x [hunk 2]
```diff
@@ -128,6 +129,16 @@ enum SCHostStorageErrorCode
...
+    HOST_AUTH_GET_RECORDING_IN_ENFORCING_MODE = 2,
```
Comment (dmkozh) [inline Stellar-contract.x]: This doesn't seem right - recording mode is not a part of the protocol and it's not possible to trigger this from a transaction. So this should probably be just a general error with an appropriate message.

File: Stellar-contract.x (hunk 1 above)
Comment (sisuresh) [reply Stellar-contract.x]: Done

File: Stellar-contract.x (hunk 2 above)
Comment (sisuresh) [reply Stellar-contract.x]: Removed HOST_AUTH_GET_RECORDING_IN_ENFORCING_MODE

**Review by dmkozh — APPROVED** [review]

**Pull request merged**